src/citrine/informatics/design_spaces/design_space.py

```diff
@@ -21,6 +24,27 @@ class DesignSpace(Module):
     name = properties.String('config.name')
     description = properties.Optional(properties.String(), 'config.description')
 
+    # TODO: This property should live in AIResourceMetadata. However, since not all modules which
+    # inherit from that class support status_detail, it must live here until they do.
+    status_detail = properties.List(properties.Object(StatusDetail), 'status_detail', default=[],
+                                    serializable=False)
+    """:List[StatusDetail]: a list of structured status info, containing the message and level"""
+
+    @property
+    @deprecated(deprecated_in="1.45.0", removed_in="2.0.0",
+                details="Please use the 'status_detail' property instead.'")
+    def status_info(self):
+        # TODO: This property should live in AIResourceMetadata. However, since not all modules
+        # which inherit from that class support status_detail, it must live here until they do.
+        """[DEPRECATED] :list[str]: human-readable explanations of the status."""
+        return [detail.msg for detail in self.status_detail]
+
+    @status_info.setter
+    def status_info(self, value):
+        # This is required to allow deserialization to work, since it tries to treat status_info as
+        # a property, as defined in AIResourceMetadata.
+        pass  # pragme: no cover
```

Comment:
Why are we not skipping this implementation, or throwing a `RuntimeError`?  A no-op setter seems like bad mojo.

Reply:
Backwards compatibility.

This was previously a property, which as a Python field, users were welcome to write to if they wanted. But as it had `serializable=False`, it wouldn't be sent to the backend. So of course, users _shouldn't_ write to it, but as it was previously allowed, we shouldn't break that capability.

Granted, you could make the argument we're changing its behavior. I wanted to do the least possible to not break the interface. But if you feel it's important for this behavior to stay the same, I can rig something up to make it work same as before if they set it.

Reply:
Makes sense.  Maybe we should be raising a Warning, perhaps a DeprecationWarning?  That would at least be noisy.

Reply:
Agreed, it definitely should.